AspNetCoreApplication/Services/UserClaimsPrincipal.cs

```diff
@@ -0,0 +1,28 @@
+﻿using AspNetCoreApplication.DTO.DTOuser;
+using System;
+using System.Collections.Generic;
+using AspNetCoreApplication.Models;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using System.IdentityModel.Tokens.Jwt;
+using AspNetCoreApplication.Extensions;
+using AspNetCoreApplication.Mappings;
+
+namespace AspNetCoreApplication.Services
+{
+    public class UserClaimsPrincipal : ClaimsPrincipal
+    {
+        public AuthenUser AuthenUser { get; private set; }
+        public UserClaimsPrincipal(ClaimsIdentity claimsIdentity) : base(claimsIdentity)
+        {
+            AuthenUser = new AuthenUser()
+            {
+                Username = claimsIdentity.GetClaimValue(ClaimTypes.Upn),
+                Name = claimsIdentity.GetClaimValue(ClaimTypes.GivenName),
+                Id = claimsIdentity.GetClaimValue(ClaimTypes.NameIdentifier),
+                Role = claimsIdentity.GetClaimValue(ClaimTypes.Role).ConvertTo<Role>()
```

Comment:
Anh ko chắc câu lệnh này hoạt động, cần test lại cẩn thận
```suggestion
                Role = claimsIdentity.GetClaimValue(ClaimTypes.Role).ConvertTo<Role>()
```

Reply:
done